### PR TITLE
chore: Adjusts release process to run tests against QA before releasing

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): [CLOUDP-220682](https://jira.mongodb.org/browse/CLOUDP-220682)

Used a [forked repository](https://github.com/AgustinBettati/terraform-provider-mongodbatlas/actions) to test logic of when acceptances test and release is run.

Scenarios that were tested:
- When providing an invalid input (1.0.0): execution of tests and release is skipped
- When doing a release (v1.0.0) or a pre release (v1.0.0-pre):
  - with skip value `true` -> tests are not run and release is made
  - with skip value empty -> tests are run
    - if test fails release job is not run
    - if test are successful release job runs

What has not been tested and will be tested on release?
- During testing in the forked repository execution of acceptance tests was replaced with a dummy job, so the call to the acceptance test workflow it not tested (was copied from working definition in [test-suite.yml#L63](https://github.com/AgustinBettati/terraform-provider-mongodbatlas/blob/testing-release-logic/.github/workflows/test-suite.yml#L63).

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [x] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: .github/workflows/release.yml
Comment (AgustinBettati) [inline .github/workflows/release.yml:43]: as of now tag is not being signed, which is the behaviour we currently have when [creating tags for our releases](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/RELEASING.md#pre-release-the-provider). Will reach out to security team to verify if we can use the gpg signing feature of this action.

File: .github/workflows/release.yml
Comment (andreaangiolillo) [inline .github/workflows/release.yml:40]: Do we want to add the logic to delete the tag in case of failure of a step after the tag creation?

File: .github/workflows/release.yml
Comment (andreaangiolillo) [inline .github/workflows/release.yml:8]: should we add a step to validate that the provided tag is correct? One common error is to forget the V

File: .github/workflows/release.yml
Comment (andreaangiolillo) [inline .github/workflows/release.yml]: why do we always skip QA tests for pre-release? I see a scenario where we want to run the QA as part of the pre-release instead of the release. The pre-release will contain all the code changes except the changelog PR.

File: .github/workflows/release.yml
Comment (AgustinBettati) [reply .github/workflows/release.yml:8]: yes this should be simple to include and has value

File: .github/workflows/release.yml
Comment (AgustinBettati) [reply .github/workflows/release.yml]: did not want to slow down the process of generating the pre release, but I do see the value in detecting an issue in the prerelease vs seeing the issue in the final release. We can leave it to run by default, and have the option of skipping if you have already triggered manually for example.

Comment (AgustinBettati) [reply .github/workflows/release.yml]: adjusted workflow and documentation.

File: .github/workflows/release.yml
Comment (andreaangiolillo) [reply .github/workflows/release.yml:43]: Could you confirm that the action is creating an annotated tag?

File: .github/workflows/release.yml
Comment (andreaangiolillo) [inline .github/workflows/release.yml:41]: I've found this action https://github.com/mathieudutour/github-tag-action (520 starts) which seems to be used more than rickstaa/action-create-tag (60 starts). Maybe we could use that one (not a strong opinion)

File: .github/workflows/release.yml
Comment (AgustinBettati) [reply .github/workflows/release.yml:43]: yes verified that annotated tags were created in the forked repo used for testing, can also be seen in the code https://github.com/rickstaa/action-create-tag/blob/main/entrypoint.sh#L79

File: .github/workflows/release.yml
Comment (AgustinBettati) [reply .github/workflows/release.yml:41]: Yes that was an option I looked into, went for https://github.com/rickstaa/action-create-tag because it had the feature of signing tags (even though we are not using it yet) which this other action did not have.

Comment (andreaangiolillo) [reply .github/workflows/release.yml:41]: it makes sense. Thanks!

File: .github/workflows/release.yml
Comment (AgustinBettati) [reply .github/workflows/release.yml:40]: This can be an improvement but not sure if worth including in this PR. This case would only happen if `ghaction-import-gpg` or `goreleaser-action` action fails, which does not seem frequent from previous releases.

Comment (andreaangiolillo) [reply .github/workflows/release.yml:40]: I'd suggest documenting what we should do in case of a failure during the go releaser step in this PR: manually delete the tag and try again to run the action. Feel free to create a ticket to add this improvement if you don't want to do it as part of this PR.

**Review by andreaangiolillo — APPROVED** [review]
LGTM. Thanks for all the follow ups

File: .github/workflows/release.yml
Comment (AgustinBettati) [reply .github/workflows/release.yml:40]: leaving created jira ticket just for reference: https://jira.mongodb.org/browse/CLOUDP-221414

**Pull request merged**
